src/hearings/resolvers/priority/priority.resolve.ts

```diff
@@ -19,15 +19,16 @@ export class PriorityResolver implements Resolve<RefDataModel[]> {
     private readonly hearingStore: Store<fromHearingStore.State>
   ) { }
 
-  public resolve(): Observable<RefDataModel[]> {
+  public resolve(route?: ActivatedRouteSnapshot): Observable<RefDataModel[]> {
     return this.getServiceId$()
       .pipe(
         switchMap(id => {
           return of(
             id ? id : this.serviceId);
         }), take(1),
         switchMap((serviceId) => {
-          return this.getReferenceData$(serviceId);
+          const category = route.data['category'] ? route.data['category'] as HearingCategory : HearingCategory.Priority;
+          return this.getReferenceData$(serviceId, category);
```

Comment:
After this is changed, I suppose you also need to add the 'data' into the route of DatePriorityHearingComponent.

Reply:
This is done but not marked it @rayliangatos